examples/trials/mnist-bohb/mnist.py

```diff
@@ -0,0 +1,250 @@
+    graph_location = tempfile.mkdtemp()
+    logger.debug('Saving graph to: %s', graph_location)
+    train_writer = tf.summary.FileWriter(graph_location)
+    train_writer.add_graph(tf.get_default_graph())
+
+    test_acc = 0.0
+    with tf.Session() as sess:
+        sess.run(tf.global_variables_initializer())
+        for i in range(params['batch_num']):
+            batch = mnist.train.next_batch(params['batch_size'])
+            mnist_network.train_step.run(feed_dict={mnist_network.images: batch[0],
+                                                    mnist_network.labels: batch[1],
+                                                    mnist_network.keep_prob: 1 - params['dropout_rate']}
+                                        )
+
+            if i % 100 == 0:
+                test_acc, test_loss = sess.run(
+                    [mnist_network.accuracy, mnist_network.loss],
+                    feed_dict={mnist_network.images: mnist.test.images,
+                               mnist_network.labels: mnist.test.labels,
+                               mnist_network.keep_prob: 1.0})
+                test_result = {
+                    "default": test_acc,
+                    "loss": test_loss
+                }
+                nni.report_intermediate_result(test_result)
+                logger.debug('test accuracy %g test loss %g', test_result["default"], test_result["loss"])
+                logger.debug('Pipe send intermediate result done.')
+
+        test_acc, test_loss = sess.run(
+            [mnist_network.accuracy, mnist_network.loss],
+            feed_dict={mnist_network.images: mnist.test.images,
+                        mnist_network.labels: mnist.test.labels,
+                        mnist_network.keep_prob: 1.0})
+        test_result = {
+            "default": test_acc,
+            "loss": test_loss
+        }
+        nni.report_final_result(test_result)
+        logger.debug('Final result is %g test loss %g', test_result["default"], test_result["loss"])
+        logger.debug('Send final result done.')
+
+def get_params():
+    ''' Get parameters from command line '''
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--data_dir", type=str, default='/tmp/tensorflow/mnist/input_data', help="data directory")
+    parser.add_argument("--dropout_rate", type=float, default=0.5, help="dropout rate")
+    parser.add_argument("--channel_1_num", type=int, default=32)
+    parser.add_argument("--channel_2_num", type=int, default=64)
+    parser.add_argument("--conv_size", type=int, default=5)
+    parser.add_argument("--pool_size", type=int, default=2)
+    parser.add_argument("--hidden_size", type=int, default=1024)
+    parser.add_argument("--learning_rate", type=float, default=1e-4)
+    parser.add_argument("--batch_num", type=int, default=2700)
+    parser.add_argument("--batch_size", type=int, default=32)
+
+    args, _ = parser.parse_known_args()
+    return args
+
+if __name__ == '__main__':
+    try:
```

Comment:
I suggest reuse the example code in hyperband. Which we could reduce one folder.

Reply:
Actually, the trial codes in hyerband and bohb are not the same, because bohb requires to return a **dict**, but it is ok to merge them into a `mnist-advisor` file. What do you think?

Reply:
Mmerge them into a mnist-advisor file now.